build.gradle

```diff
@@ -86,3 +86,18 @@ dependencies {
             "com.github.tomakehurst:wiremock-jre8:$wireMockVersion"
     testImplementation 'org.mockito:mockito-inline:3.3.0'
 }
+
+task shadowBugWorkaround(type: Jar) {
+    archiveBaseName = 'nested-gradle-jar'
+    from files(file("${rootDir}/gradle/wrapper/gradle-wrapper.jar"))
+    destinationDir file('build/shadow-bug-workaround')
+}
+
+shadowJar {
+    dependsOn shadowBugWorkaround
+    zip64 = true
+    // Shadow plugin bug: explodes the nested jar.
+    // Workaround: double-nest the jar.
+    // Refer https://github.com/johnrengelman/shadow/issues/111
+    from shadowBugWorkaround
+}
```

Comment:
As we have a `gradle-wrapper.jar` in the resources which is used during project generation, Shadow explodes this and causes a NPE by default, this workaround causes the jar file to be embedded normally as a resource.